Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Link.swift

```diff
@@ -26,24 +43,117 @@ struct UserProfileLink: View
                     }
                 }
                 
-                Text(user.name)
+                /* Disabled icons for now */
+                if let flair = calculateFlair() {
+                    if let flairSystemIcon = flair.systemIcon {
+                        Image(systemName: flairSystemIcon).foregroundColor(flair.color)
+                    }
+                }
+                
+                
+                // User display name if one exists
+                Text(user.displayName ?? user.name)
                     .minimumScaleFactor(0.01)
                     .lineLimit(1)
             }
-            .if(user.admin)
+            .if(calculateFlair() != nil)
             { viewProxy in
                 viewProxy
-                    .foregroundColor(.red)
+                    .foregroundColor(calculateFlair()!.color).bold()
             }
-            .if(user.botAccount == true)
-            { viewProxy in
-                viewProxy
-                    .foregroundColor(.indigo)
+        }
+    }
+    
+    struct UserProfileLinkFlair {
+        var color: Color
+        var systemIcon: String? = nil
+    }
+    
+    private func calculateFlair() -> UserProfileLinkFlair? {
+        if let userServer = user.actorId.host() {
+            if UserProfileLink.developerNames.contains(where: { $0 == "\(userServer)\(user.actorId.path())" }) {
+                return UserProfileLink.flairDeveloper 
             }
-            .if(user.name == "lFenix")
-            { viewProxy in
-                viewProxy
-                    .foregroundColor(.yellow)
+        }
+        if user.admin {
+            return UserProfileLink.flairAdmin
+        }
+        if user.botAccount {
+            return UserProfileLink.flairBot
+        }
+        if commentContext != nil {
+            if commentContext!.distinguished {
```

Comment:
you can do `if let commentContext` and it essentially does nil check and the unwrapping, safely